src/jemalloc.c

```diff
@@ -1382,6 +1425,18 @@ malloc_conf_init(sc_data_t *sc_data, unsigned bin_shard_sizes[SC_NBINS]) {
 	atomic_store_b(&log_init_done, true, ATOMIC_RELEASE);
 }
 
+static void
+malloc_conf_init(sc_data_t *sc_data, unsigned bin_shard_sizes[SC_NBINS]) {
+	const char *opts_cache[4] = {NULL, NULL, NULL, NULL};
+	char buf[PATH_MAX + 1];
+	const char empty_opts = '\0';
```

Comment:
Let's get rid of this hack; and pass a explicit flag "use_cached_opts" to the helper -- first call will be false which reads the opts and fill the cache.